internal/app/bootkube/main.go

```diff
@@ -52,6 +52,13 @@ func run() error {
 	}
 
 	defer func() {
+		// We want to cleanup the manifests directory only if bootkube fails.
+		if err != nil {
+			if err = os.RemoveAll(constants.ManifestsDirectory); err != nil {
```

Comment:
I don't think this works as `err` is set on line 49 only, and `defer` block is later than `return err` on line 51